commit ef3d8b2b5ea9af14360074bf79d7e8c4340b68cb
Author: C. Boemann <cbo@boemann.dk>
Date:   Sun Nov 18 08:44:32 2012 +0100

    Apply patch that fixes build of videoshape
    
    Thanks to Raúl Sánchez Siles
    
    CCMAIL: rasasi78@gmail.com

diff --git a/plugins/videoshape/CMakeLists.txt b/plugins/videoshape/CMakeLists.txt
index a9f89ef..66c1376 100644
--- a/plugins/videoshape/CMakeLists.txt
+++ b/plugins/videoshape/CMakeLists.txt
@@ -1,5 +1,13 @@
 include_directories( ${PHONON_INCLUDE_DIR} )
 
+if(PHONON_FOUND_EXPERIMENTAL)
+   set(SHOULD_BUILD_THUMBNAIL TRUE)
+   add_definitions( -DSHOULD_BUILD_THUMBNAIL )
+else(PHONON_FOUND_EXPERIMENTAL)
+    set(SHOULD_BUILD_THUMBNAIL FALSE)
+endif(PHONON_FOUND_EXPERIMENTAL)
+
+
 ########### Flake Plugin library ###############
 
 SET (videoshape_LIB_SRCS
@@ -15,16 +23,20 @@ SET (videoshape_LIB_SRCS
     VideoToolFactory.cpp
     ChangeVideoCommand.cpp
     SelectVideoWidget.cpp
-    VideoThumbnailer.cpp
 )
 
+if(SHOULD_BUILD_THUMBNAIL)
+    set(videoshape_LIB_SRCS ${videoshape_LIB_SRCS} VideoThumbnailer.cpp)
+endif(SHOULD_BUILD_THUMBNAIL)
+
 kde4_add_ui_files(videoshape_LIB_SRCS forms/VideoToolWidget.ui)
 kde4_add_plugin(videoshape ${videoshape_LIB_SRCS})
 
-target_link_libraries(videoshape komain phononexperimental
+target_link_libraries(videoshape komain
     ${KDE4_KFILE_LIBRARY}
-    ${KDE4_PHONON_LIBRARY}
+    ${PHONON_LIBS}
 )
+
 install(TARGETS videoshape DESTINATION ${PLUGIN_INSTALL_DIR})
 
 ########### install files ###############
diff --git a/plugins/videoshape/VideoShape.cpp b/plugins/videoshape/VideoShape.cpp
index ff5b4ae..d221098 100644
--- a/plugins/videoshape/VideoShape.cpp
+++ b/plugins/videoshape/VideoShape.cpp
@@ -27,7 +27,9 @@
 #include <VideoEventAction.h>
 #include <VideoCollection.h>
 #include <VideoData.h>
+#ifdef SHOULD_BUILD_THUMBNAIL
 #include <VideoThumbnailer.h>
+#endif
 #include <KoShapeLoadingContext.h>
 #include <KoOdfLoadingContext.h>
 #include <KoShapeSavingContext.h>
@@ -47,7 +49,9 @@ VideoShape::VideoShape()
     , m_videoEventAction(new VideoEventAction(this))
     , m_icon(koIconName("video-x-generic"))
     , m_oldVideoData(0)
+#ifdef SHOULD_BUILD_THUMBNAIL
     , m_thumbnailer(new VideoThumbnailer())
+#endif
 {
     setKeepAspectRatio(true);
     addEventAction(m_videoEventAction);
@@ -55,7 +59,9 @@ VideoShape::VideoShape()
 
 VideoShape::~VideoShape()
 {
+#ifdef SHOULD_BUILD_THUMBNAIL
     delete m_thumbnailer;
+#endif
 }
 
 void VideoShape::paint(QPainter &painter, const KoViewConverter &converter, KoShapePaintingContext &)
@@ -63,7 +69,7 @@ void VideoShape::paint(QPainter &painter, const KoViewConverter &converter, KoSh
     QRectF pixelsF = converter.documentToView(QRectF(QPointF(0,0), size()));
 
     VideoData *currentVideoData = videoData();
-
+#ifdef SHOULD_BUILD_THUMBNAIL
     if (currentVideoData && currentVideoData != m_oldVideoData) {
         //generate thumbnails
         m_oldVideoData = currentVideoData;
@@ -79,6 +85,13 @@ void VideoShape::paint(QPainter &painter, const KoViewConverter &converter, KoSh
     } else {
         painter.drawImage(pixelsF, thumnailImage);
     }
+#else
+    painter.fillRect(pixelsF, QColor(Qt::gray));
+    painter.setPen(QPen());
+    painter.drawRect(pixelsF);
+
+    m_icon.paint(&painter, pixelsF.toRect());
+#endif
 }
 
 void VideoShape::saveOdf(KoShapeSavingContext &context) const
diff --git a/plugins/videoshape/VideoShape.h b/plugins/videoshape/VideoShape.h
index 059220a..12a8545 100644
--- a/plugins/videoshape/VideoShape.h
+++ b/plugins/videoshape/VideoShape.h
@@ -33,7 +33,9 @@
 class VideoCollection;
 class VideoEventAction;
 class VideoData;
+#ifdef SHOULD_BUILD_THUMBNAIL
 class VideoThumbnailer;
+#endif
 
 class QImage;
 
@@ -76,7 +78,9 @@ signals:
 private:
     VideoCollection *m_videoCollection;
     VideoEventAction *m_videoEventAction;
+#ifdef SHOULD_BUILD_THUMBNAIL
     VideoThumbnailer *m_thumbnailer;
+#endif
     VideoData *m_oldVideoData;
     KIcon m_icon;
     QRectF m_playIconArea;
